WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingLaunchStoreView.swift

```diff
@@ -50,8 +51,14 @@ struct StoreOnboardingLaunchStoreView: View {
                         await viewModel.launchStore()
                     }
                 }
-                .buttonStyle(PrimaryLoadingButtonStyle(isLoading: viewModel.isLaunchingStore))
+                .if((viewModel.state == .launchingStore || viewModel.state == .readyToPublish)) {
+                    $0.buttonStyle(PrimaryLoadingButtonStyle(isLoading: viewModel.state == .launchingStore))
+                }
+                .if(viewModel.state == .needsPlanUpgrade) {
+                    $0.buttonStyle(PrimaryButtonStyle()).disabled(true)
+                }
```

Comment:
Nit: this can be simplified by using just `PrimaryLoadingButtonStyle` - since when loading state is false, this style is the same as `PrimaryButtonStyle`. We can do this instead:

```suggestion
               .buttonStyle(PrimaryLoadingButtonStyle(isLoading: viewModel.isLaunchingStore))
               .disabled(viewModel.state == .needsPlanUpgrade)
```

Also, should we add a variable in the view model for `needsPlanUpgrade` so we don't need to compare the state twice?

Reply:
Done in 5bdf8037abd798dec170c568c9964141894c87b1

> Also, should we add a variable in the view model for needsPlanUpgrade so we don't need to compare the state twice?

I prefer to avoid introducing variables and maintain them and instead run the view using a single state value.